op-supervisor/supervisor/backend/l1access/l1_accessor.go

```diff
@@ -159,8 +162,33 @@ func (p *L1Accessor) onFinalized(ctx context.Context, ref eth.L1BlockRef) {
 }
 
 func (p *L1Accessor) onLatest(ctx context.Context, ref eth.L1BlockRef) {
+	// Stop if the block is the same or older than the tip
+	if ref.Number == p.tipHeight && ref.Hash == p.tipHash {
+		p.log.Warn("L1 block is the same as the tip", "ref", ref)
+		return
+	}
+	if ref.Number < p.tipHeight {
+		p.log.Warn("L1 block is older than the tip", "ref", ref)
+		return
+	}
+
+	// Update the tip
 	p.tipHeight = ref.Number
+	p.tipHash = ref.Hash
 	p.log.Info("Updated latest known L1 block", "ref", ref)
+
+	// Check if the block is the next one and if it is, check for a reorg
+	if ref.Number != p.tipHeight+1 {
+		p.log.Warn("L1 block is not the next one", "ref", ref)
+		return
```

Comment:
If for some reason, the L1 tip reference changed by more than one *and* experienced a reorg at the same time, the database would never notice the L1 reorg. One such case might be if the supervisor went down and several of its recorded L1 blocks got reorged out while it was down.

I think this means we should check for consistency to the tip at all times, even when the gap is more than one. To accomplish this, you'd walk forward from the current `tipRef` to `ref.Number`, comparing parent hashes the whole way. If a lineage can't be drawn from the current tip to the new tip, *then* you can issue a reorg.

Reply:
The changes to L1 reorg handling include a more robust detection.